plugins/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/IngestGeoIpPlugin.java

```diff
@@ -119,4 +119,75 @@ public void close() throws IOException {
         }
     }
 
+    /**
+     * The in-memory cache for the geoip data. There should only be 1 instance of this class..
+     * This cache differs from the maxmind's {@link NodeCache} such that this cache stores the deserialized Json objects to avoid the
+     * cost of deserialization for each lookup (cached or not). This comes at slight expense of higher memory usage, but significant
+     * reduction of CPU usage.
+     */
+    class GeoIpCache {
```

Comment:
can't this be static?

Reply:
I don't think it makes sense to allow instances of this class to exist outside of the IngestGeoIpPlugin which it is nested. Hence not static.

Reply:
Static is about whether there is state needed from the outer class. This is still package private, so I don't see it as a problem for someone trying to construct this (especially since it is all within the geoid plugin project anyways.

Reply:
> Static is about whether there is state needed from the outer class.

Non static inner classes are also an effective for defining the logical scope to which something belongs (if even if the outer state's class is not needed). In this case you *need* a instance of a plugin to instantiate a cache. This helps to enforce the relationship between these two classes. IMO this helps to enforce this cache is not intended to be used by any other class (especially the by processor class...we don't a cache per processor instance) since it's logical scope is per-plugin.  If I could, it would be fully private, but for testing reasons it is not. 

This is mostly academic and I don't have strong opposition, just wanted to describe my rationale in more detail.  

I'll defer to your opinion and change this (and below) to static.